src/App.tsx

```diff
@@ -17,18 +56,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onFilterChange={handleFilterChange}
+                onQueryChange={handleQueryChange}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    onChoose={handleTodoSelect}
+                    selectedTodoId={selectedTodoId}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      <TodoModal
```

Comment:
Let's remove the modal from DOM when it closes
```suggestion
  {  selectedTodoId && <TodoModal
```